lapack/gonum/dormbr.go

```diff
@@ -45,40 +45,43 @@ import (
 //
 // Dormbr is an internal routine. It is exported for testing purposes.
 func (impl Implementation) Dormbr(vect lapack.ApplyOrtho, side blas.Side, trans blas.Transpose, m, n, k int, a []float64, lda int, tau, c []float64, ldc int, work []float64, lwork int) {
-	if side != blas.Left && side != blas.Right {
-		panic(badSide)
-	}
-	if trans != blas.NoTrans && trans != blas.Trans {
-		panic(badTrans)
-	}
-	if vect != lapack.ApplyP && vect != lapack.ApplyQ {
-		panic(badApplyOrtho)
-	}
 	nq := n
 	nw := m
 	if side == blas.Left {
 		nq = m
 		nw = n
 	}
-	if vect == lapack.ApplyQ {
-		checkMatrix(nq, min(nq, k), a, lda)
-	} else {
-		checkMatrix(min(nq, k), nq, a, lda)
-	}
-	if len(tau) < min(nq, k) {
-		panic(badTau)
-	}
-	checkMatrix(m, n, c, ldc)
-	if len(work) < lwork {
-		panic(shortWork)
-	}
-	if lwork < max(1, nw) && lwork != -1 {
+	applyQ := vect == lapack.ApplyQ
+	switch {
+	case !applyQ && vect != lapack.ApplyP:
+		panic(badApplyOrtho)
+	case side != blas.Left && side != blas.Right:
+		panic(badSide)
+	case trans != blas.NoTrans && trans != blas.Trans:
+		panic(badTrans)
+	case m < 0:
+		panic(mLT0)
+	case n < 0:
+		panic(nLT0)
+	case k < 0:
+		panic(kLT0)
+	case applyQ && lda < max(1, min(nq, k)):
```

Comment:
Just a note; no change necessary.

This could be written as `applyQ && lda < max(1, min(nq, k)), !applyQ && lda < max(1, nq)`. However, I think it is better as you have it since the panic line gives the reason for the failure where the flattened approach would require digging.

Reply:
Ok, I will leave it unchanged.